src/hostname/hostnamed.c

```diff
@@ -729,19 +729,20 @@ static int context_write_data_static_hostname(Context *c) {
 
         assert(c);
 
+        const char *hostname_path = etc_hostname();
```

Comment:
I'd like to drop such tentative variables.
Please cache the value in `etc_hostname()` and friends.